### PR TITLE
plugins/lsp: add svelte language server

### LINKED ISSUE #497: Svelte language server
**Plugin**: nodePackages.svelte-language-server
**Plugin repository**: https://github.com/sveltejs/language-tools#readme
**Is the plugin on nixpkgs**: https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/development/node-packages/node-packages.nix#L143184

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for the [svelte language server](https://github.com/sveltejs/language-tools/tree/master/packages/language-server).

Fixes #497 

@rupurt, if you have the possibility to test it on your setup, it would be great !

### CODE REVIEW

**Comment by rupurt** [comment]
Looks good on my machine :pray:

File: plugins/lsp/language-servers/svelte.nix [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+          hover = {
+            enable = helpers.defaultNullOpts.mkBool true ''
+              Enable hover info for CSS.
+            '';
+          };
+
+          completions = {
+            enable = helpers.defaultNullOpts.mkBool true ''
+              Enable completions for TypeScript.
```
Comment (traxys) [inline plugins/lsp/language-servers/svelte.nix:89]: completions for CSS

File: plugins/lsp/language-servers/svelte.nix [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+        svelte = {
+          enable = helpers.defaultNullOpts.mkBool true "Enable the Svelte plugin.";
+
+          diagnostics = {
+            enable = helpers.defaultNullOpts.mkBool true ''
+              Enable diagnostic messages for Svelte.
+            '';
+
+            compilerWarnings = helpers.mkNullOrOption (with types; attrsOf str) ''
```
Comment (traxys) [inline plugins/lsp/language-servers/svelte.nix:168]: compilerwarnings is not in diagnostics

**Review by traxys — CHANGES_REQUESTED** [review]

**Review by traxys — APPROVED** [review]

**Comment by rupurt** [comment]
FWIW, working great on main. That's again y'all!

**Pull request merged**